core/server/services/routing/TaxonomyRouter.js

```diff
@@ -70,7 +70,7 @@ class TaxonomyRouter extends ParentRouter {
     }
 
     getResourceType() {
-        return RESOURCE_CONFIG.QUERY[this.taxonomyKey].resource;
+        return RESOURCE_CONFIG.QUERY[this.taxonomyKey].alias;
```

Comment:
This needs cleanup. Super confusing why we have alias and resource.
Furthermore we should forward `routerOptions.query`. This object should contain the information which controller to query....based on the api version?